webapp/packages/core-connections/src/ConnectionNavigatorViewSettings.ts

```diff
@@ -8,9 +8,12 @@
 
 import type { NavigatorSettingsInput } from '@cloudbeaver/core-sdk';
 
-type NavigatorView = 'simple' | 'advanced';
+export type NavigatorView = 'simple' | 'advanced' | 'custom';
+export type NavigatorViewSettingsKeys = keyof NavigatorSettingsInput;
 
-export const CONNECTION_NAVIGATOR_VIEW_SETTINGS: Record<NavigatorView, NavigatorSettingsInput> = {
+type NavigatorViewSettings = Record<Exclude<NavigatorView, 'custom'>, NavigatorSettingsInput>;
+
+export const CONNECTION_NAVIGATOR_VIEW_SETTINGS: NavigatorViewSettings = {
```

Comment:
seems like we should create `DEFAULT_NAVIGATOR_VIEW_SETTING` with type `NavigatorSettingsInput`